vignettes/containerit.Rmd

```diff
@@ -70,177 +134,202 @@ session <- clean_session(exp, echo = TRUE)
-addInstruction(dockerfile_object) <- label
+Label_LabelSchema <- LabelSchemaFactory()
+label <- Label_LabelSchema(name = "ImageName", description = "Description of the image", build_date = Sys.time())
+addInstruction(labeled_dockerfile) <- label
 ```
 
-Put session info output into a label as plain text (optional use as_json = TRUE):
+You can also put session information, using either base R or `devtools`, into a label as plain text or as json:
 
 ```{r}
-addInstruction(dockerfile_object) <- Label_SessionInfo(clean_session())
+addInstruction(labeled_dockerfile) <- Label_SessionInfo(session = clean_session())
+addInstruction(labeled_dockerfile) <- Label_SessionInfo(session = devtools::session_info(), as_json = TRUE)
 ```
 
+The resulting Dockerfile with all the labels:
 
-Result:
-
-```{r}
-print(dockerfile_object)
+```{r, comment=NA}
+print(labeled_dockerfile)
 ```
 
 ## 5. Further costumization 
 
-
-- Use costum base image (rocker-stack) and costum R version
+The `dockerfile()` function allows further customization regarding the R version or the used base image (cf. Rocker stack).
+Note that while choosing an R version for the Dockerfile explicitly is possible, the session to generate the required information (i.e. which packages are attached etc.) is still running the R version of the generating machine.
 
 ```{r}
-dockerfile_object <- dockerfile(from = NULL, r_version = "3.1.0", silent = TRUE)
-print(dockerfile_object@image)
-dockerfile_object <- dockerfile(from = NULL, image = "rocker/geospatial", silent = TRUE)
-print(dockerfile_object@image)
-dockerfile_object <- dockerfile(from = NULL, image = "rocker/verse:3.0.0", silent = TRUE)
-print(dockerfile_object@image)
+df_custom <- dockerfile(from = NULL, r_version = "3.1.0", silent = TRUE)
+print(df_custom@image)
+df_custom <- dockerfile(from = NULL, image = "rocker/geospatial", silent = TRUE)
+print(df_custom@image)
+df_custom <- dockerfile(from = NULL, image = "rocker/verse:3.0.0", silent = TRUE)@image
+print(df_custom@image)
 ```
 
+## 6. CLI
+
+A command line interface to the package functions is also available for Linux based on [docopt.R](https://github.com/docopt/docopt.R).
+This allows integration into workflows and tools written in other programming languages than R.
+
+You can make the command `containerit` available on your maching by linking the R script file delivered with the package as follows:
 
-CLI
-installation on Linux via `ln -s $(Rscript -e "cat(system.file(\"cli/container_it.R\", package=\"containeRit\"))") /usr/local/bin/container_it.R`
+`ln -s $(Rscript -e "cat(system.file(\"cli/container_it.R\", package=\"containeRit\"))") /usr/local/bin/containerit`
 
 CLI Examples:
+
 ```
+  containerit --help
+  
   # runs the first R markdown or R script file locally 
   # prints Dockerfile without writing a file
-  container_it.R dir -p --no-write  
+  containerit dir -p --no-write  
```

Comment:
We can only do this, if we also rename the file in **container_it.R** to just **containerit**. Then we must also update the install command above.

Since it is also a real working R script, it would also make sense to keep the .R file ending. But well, maybe a matter of taste...